c/secp256k1_helper.h

```diff
@@ -0,0 +1,88 @@
+#ifndef CKB_SECP256K1_HELPER_H_
+#define CKB_SECP256K1_HELPER_H_
+
+#include "ckb_syscalls.h"
+#include "secp256k1_data_info.h"
+
+#define CKB_SECP256K1_HELPER_ERROR_LOADING_DATA -101
+#define CKB_SECP256K1_HELPER_ERROR_ILLEGAL_CALLBACK -102
+#define CKB_SECP256K1_HELPER_ERROR_ERROR_CALLBACK -103
+
+/*
+ * We are including secp256k1 implementation directly so gcc can strip
+ * unused functions. For some unknown reasons, if we link in libsecp256k1.a
+ * directly, the final binary will include all functions rather than those used.
+ */
+#define HAVE_CONFIG_H 1
+#define USE_EXTERNAL_DEFAULT_CALLBACKS
+#include <secp256k1.c>
+
+void secp256k1_default_illegal_callback_fn(const char* str, void* data) {
+  (void)str;
+  (void)data;
+  ckb_exit(CKB_SECP256K1_HELPER_ERROR_ILLEGAL_CALLBACK);
+}
+
+void secp256k1_default_error_callback_fn(const char* str, void* data) {
+  (void)str;
+  (void)data;
+  ckb_exit(CKB_SECP256K1_HELPER_ERROR_ERROR_CALLBACK);
+}
+
+/*
+ * data should at least be CKB_SECP256K1_DATA_SIZE big
+ * so as to hold all loaded data.
+ */
+int ckb_secp256k1_custom_verify_only_initialize(secp256k1_context* context,
+                                                void* data) {
+  size_t index = 0;
+  int running = 1;
+  while (running && index < SIZE_MAX) {
+    volatile uint64_t len = 32;
+    uint8_t hash[32];
+
+    int ret = ckb_load_cell_by_field(hash, &len, 0, index, CKB_SOURCE_CELL_DEP,
+                                     CKB_CELL_FIELD_DATA_HASH);
+    switch (ret) {
+      case CKB_ITEM_MISSING:
+        break;
+      case CKB_SUCCESS:
+        if (memcmp(ckb_secp256k1_data_hash, hash, 32) == 0) {
```

Comment:
Should we support `load_cell_by_hash` syscall?

Reply:
I think that will be a separate problem, since if we want to support fetching by hashes, there should be a lot of things we can consider: load cell/script/cell data by hash.